Support_Your_Locals/Controllers/AuthController.cs

```diff
@@ -41,27 +41,51 @@ public async Task<ActionResult> SignUp(UserRegisterModel register)
                 register.AlreadyExists = exists;
                 if (!exists)
                 {
-                    new RNGCryptoServiceProvider().GetBytes(salt = new byte[16]);
-                    var pbkdf2 = new Rfc2898DeriveBytes(register.Passhash, salt, 100000);
-                    byte[] hash = pbkdf2.GetBytes(20);
-                    byte[] hashBytes = new byte[36];
-                    Array.Copy(salt, 0, hashBytes, 0, 16);
-                    Array.Copy(hash, 0, hashBytes, 16, 20);
-                    string savedPasswordHash = Convert.ToBase64String(hashBytes);
+                    
                     userRepository.AddUser(new User
                     {
                         Name = register.Name,
                         Surname = register.Surname,
                         BirthDate = register.BirthDate,
                         Email = register.Email,
-                        Passhash = savedPasswordHash
+                        Passhash = PassHash(register.Passhash)
                     });
                     return Redirect("/");
                 }
                 return View(register);
             }
             return View();
         }
+        public byte[] salt = new byte[16];
+
```

Comment:
Decorate with NotAction then, because it will be assumed it is an action.

Reply:
I cant, because [NonAction] is valid on 'method' declarations